Fundamentals/Backoffice/Property-Editors/Built-in-Property-Editors/Block-Grid-Editor/index.md

```diff
@@ -0,0 +1,625 @@
+
+If you are using ModelsBuilder, the example will look like this:
+
+Example:
+
+```csharp
+@await Html.GetBlockGridHtmlAsync(Model.MyGrid)
+```
+
+To use the `GetBlockGridHtmlAsync()` method you will need to create a Partial View for each Block Type. The Partial View must be named using the alias of the Element Type that is being used as Content Model for the Block Type.
+
+These Partial View files need to go into the `Views/Partials/BlockGrid/Components/` folder.
+
+Example: `Views/Partials/BlockGrid/Components/MyElementTypeAliasOfContent.cshtml`.
+
+The Partial Views will receive a model of type `Umbraco.Core.Models.Blocks.BlockGridItem`. This model contains `Content` and `Settings` from your block, as well as the configured `RowSpan`, `ColumnSpan`, `ForceLeft`, `ForceRight` and `Areas` of the block.
+
+#### Rendering the block areas
+
+> TODO: align area semantics when the rest of the article is written
+
+The Partial View for the block is responsible for rendering its own block areas. This is done using another built-in rendering mechanism:
+
+```csharp
+@await Html.GetBlockGridItemAreasHtmlAsync(Model)
+```
+
+Here you will need to create a Partial View for each Block Type within the block area. For the name, use the alias of the Element Type that is being used as Content Model for the Block Type.
+
+These Partial Views must be placed in the same folder as before, (`Views/Partials/BlockGrid/Components/`), and will receive a model of type `Umbraco.Core.Models.Blocks.BlockGridItem`.
+
+#### Putting it all together
+
+The following is an example of a Partial View for a Block Type. It is important that the `MyElementTypeAliasOfContent`and `MyElementTypeAliasOfSettings` corresponds with the selected Element Type aliases for the given Content and Settings in your block.
+
+```csharp
+@inherits Umbraco.Cms.Web.Common.Views.UmbracoViewPage<Umbraco.Cms.Core.Models.Blocks.BlockGridItem>;
+@using ContentModels = Umbraco.Cms.Web.Common.PublishedModels;
+@{
+    var content = (ContentModels.MyElementTypeAliasOfContent)Model.Content;
+    var settings = (ContentModels.MyElementTypeAliasOfSettings)Model.Settings;
+}
+
+@* Render the value of field with alias 'heading' from the Element Type selected as Content section *@
+<h1>@content.Value("heading")</h1>
+
+@* Render the block areas *@
+@await Html.GetBlockGridItemAreasHtmlAsync(Model)
+```
+
+You can simplify the property rendering using ModelsBuilder by replacing `@content.Value("heading")` with `@content.Heading`.
+
+### 2. Build your own rendering
+
+The built-in value converter for the Block Grid lets you use the block data as you like. Call the `Value<T>` method with a type of `BlockGridModel` to have the stored value returned as a `BlockGridModel` instance.
+
+`BlockGridModel` contains the Block Grid configuration (like the number of columns as `GridColumns`) whilst also being an implementation of `IEnumerable<BlockGridItem>` (see details for `BlockGridItem` above).
```

Comment:
"See details for `BlockGridIte,` above".

Which section is this referring to? The section above this, is about another type of rendering - but, is that what this is referring to? 🤔

Reply:
Hmm, not sure. I think it should make sense when reading. If BlockGridItem isn't described above then that's a mistake. Whether it's needed to describe, or not, must be best for someone with C# skill to judge.
Do you have someone to verify what is needed?

Reply:
We do not, unfortunately.

We'll have another look at the structure, once the PR is out of draft 💪 
I might very well have missed this!